modules/homa-lite/src/lib.rs

```diff
@@ -193,17 +193,14 @@ pub mod module {
 				.expect("Max rewards cannot be above 100%; qed");
 
 			// All checks pass. Proceed with Xcm transfer.
-			let xcm_result = T::XcmTransfer::transfer(
+			T::XcmTransfer::transfer(
 				who.clone(),
 				staking_currency,
 				amount,
 				T::SovereignSubAccountLocation::get(),
 				Self::xcm_dest_weight(),
-			)?;
-			ensure!(
-				matches!(xcm_result, Outcome::Complete(_)),
-				Error::<T>::XcmTransferFailed
-			);
+			)
+			.map_err(|_| Error::<T>::XcmTransferFailed)?;
```

Comment:
do we need to map the error (and drop original error), or we can throw the original error?

Reply:
Right, we can throw the original error now, as `transfer` now returns a dispatch error. Will update.